upf/upf_pfcp_api.c

```diff
@@ -1041,27 +1246,89 @@ handle_update_pdr (upf_session_t * sx, pfcp_update_pdr_t * update_pdr,
     update->precedence = pdr->precedence;
     update->pdi.src_intf = pdr->pdi.source_interface;
 
-    if (ISSET_BIT (pdr->pdi.grp.fields, PDI_F_TEID))
+    if (ISSET_BIT (pdr->pdi.grp.fields, PDI_UE_IP_ADDRESS))
       {
-	update->pdi.fields |= F_PDI_LOCAL_F_TEID;
-	/* TODO validate TEID and mask */
-	update->pdi.teid = pdr->pdi.f_teid;
+        update->pdi.fields |= F_PDI_UE_IP_ADDR;
+        update->pdi.ue_addr = pdr->pdi.ue_ip_address;
+
+        if (!ISSET_BIT (pdr->pdi.grp.fields, PDI_SDF_FILTER) &&
+            !ISSET_BIT (pdr->pdi.grp.fields, PDI_APPLICATION_ID))
+          {
+            /* neither SDF, nor Application Id, generate a wildcard
+               ACL to make ACL scanning simpler */
+            update->pdi.fields |= F_PDI_SDF_FILTER;
+            vec_reset_length (update->pdi.acl);
+            vec_add1 (update->pdi.acl, wildcard_acl);
+          }
```

Comment:
moving this code block around makes the change harder to read. It could be just a diff fragment, but could you try to keep the block in place?

Reply:
I've lifted this code block up to have UE IP information already parsed so I could then fill created_pdrs response with both fteid and UE IP.

Reply:
UE IP is only part of `Created PDR` when UP IP assignment by UP function was requested. Since we don't support that feature (yet), including the UE IP is not needed.